lib/cocoapods-search/command/search.rb

```diff
@@ -13,26 +13,37 @@ class Search < Command
         CLAide::Argument.new('QUERY', true),
       ]
 
+      def self.all_platforms
+        Specification::PLATFORMS.map do |platform|
+          platform.to_s
+        end
+      end
+
+      def all_platforms
+        self.class.all_platforms
+      end
+
       def self.options
-        [
+        options = [
           ['--regex',   'Interpret the `QUERY` as a regular expression'],
           ['--full',    'Search by name, summary, description, and authors'],
           ['--stats',   'Show additional stats (like GitHub watchers and forks)'],
-          ['--ios',     'Restricts the search to Pods supported on iOS'],
-          ['--osx',     'Restricts the search to Pods supported on OS X'],
-          ['--watchos', 'Restricts the search to Pods supported on Watch OS'],
           ['--web',     'Searches on cocoapods.org'],
-        ].concat(super.reject { |option, _| option == '--silent' })
+        ]
+        options += all_platforms.map do |platform|
+          ["--#{platform}",     "Restricts the search to Pods supported on #{platform}"]
```

Comment:
Yeah I thougth about the same. Do you have any suggestions like adding a method to Specification which gives pretty names?